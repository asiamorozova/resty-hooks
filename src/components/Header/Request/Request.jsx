import React from 'react';
import PropTypes from 'prop-types';




const Request = ({ url, method, textBody, onChange, onSubmit }) => (
  <div>
    <form onSubmit={onSubmit}>
      
      <label> URL: <input type="text" name="url" value={url} onChange={onChange}></input>
      </label>
      
      <div><label> <input type="radio" name="method" value="get"vonChange={onChange}/> GET 
      </label>

      <label><input type="radio" name="method" value="post"vonChange={onChange}/> POST 
      </label>

      <label><input type="radio" name="method" value="put" onChange={onChange}/> PUT 
      </label>

      <label><input type="radio" name="method" value="patch" onChange={onChange}/>PATCH 
      </label>

      <label><input type="radio" name="method" value="delete" onChange={onChange}/> DELETE 
      </label>
      </div>

      <textarea placeholder="JSON Body" name="textBody" value={textBody} onChange={onChange}>
      </textarea>

      <button type="submit">Submit</button>

    </form>
  </div>

);
Request.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  textBody: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired


};

export default Request;
