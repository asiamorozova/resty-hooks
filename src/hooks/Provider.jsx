import React, { createContext, useState, useContext } from 'react';
import { makeFetch } from '../services/makeFetch';
import { useLocalStorage } from './localStorage';

const Context = createContext;

export const RestyProvider = ({ children }) => {
  const [url, setUrl] = useState('');
  const [method, setMethod] = useState('GET');
  const [body, setBody] = useState('');
  const [response, setResponse] = useState({});
  const [history, setHistory] = useLocalStorage('history', []);

  const onChange = ({ target }) => {
    if(target.name === 'url') setUrl(target.value);
    if(target.name === 'method') setMethod(target.value);
    if(target.name === 'body') setBody(target.value);

  };

  const onSubmit = event => {
    event. preventDefault();
    makeFetch(url, method, body)
      .then(response => {
        setResponse(response);
        setHistory(prevHistory => [{ url, method }, ...prevHistory]);
      });
  };

  return (
    <Context.Provider value={{ url, method, body, onChange, onSubmit, response, history }}>
      {children}
    </Context.Provider>
  );
};

export const useResty = () => {
  const context = useContext(Context);
  return context;
};


