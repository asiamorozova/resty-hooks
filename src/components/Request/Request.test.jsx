import React from 'react';
import { shallow } from 'enzyme';
import Request from './Request.jsx';
import { RestyProvider } from '../../hooks/Provider';

describe('Request', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <RestyProvider>
        <Request
          onSubmit={() => {}}
          onChange={() => {}}
          url="https://www.placecage.com/c/240/420"
          method="get"
          body="body" />
      </RestyProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});

