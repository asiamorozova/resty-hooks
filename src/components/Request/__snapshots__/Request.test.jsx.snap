// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request matches a snapshot 1`] = `
<Component
  value={
    Object {
      "body": "",
      "history": Array [],
      "method": "GET",
      "onChange": [Function],
      "onSubmit": [Function],
      "response": Object {},
      "url": "",
    }
  }
>
  <Request
    body="body"
    method="get"
    onChange={[Function]}
    onSubmit={[Function]}
    url="https://www.placecage.com/c/240/420"
  />
</Component>
`;
